tests/test_adders_listed_value_adder.py

```diff
@@ -104,40 +144,56 @@ def test__mark_value_as_listed_in_feature_profiles(test_adder):
         ), f"File {file.name} is not supposed to be changed"
 
         print(f"TEST: checking amended feature profile {file.name}")
-
         check_existence_of_output_csv_file_and_compare_with_gold_standard(
             output_file=file,
             gold_standard_file=GS_DIR_WITH_FEATURE_PROFILES_AFTER_ADDITION / file.name,
         )
 
 
-def test_add_listed_value_throws_exception_with_empty_args(test_adder):
-    for bad_set_of_values in (
-        {"feature_id": "", "new_value_en": "Value", "new_value_ru": "Значение"},
-        {"feature_id": "A-1", "new_value_en": "", "new_value_ru": "Значение"},
-        {"feature_id": "A-1", "new_value_en": "Value", "new_value_ru": ""},
-    ):
-        with pytest.raises(ListedValueAdderError, match="None of the passed strings can be empty"):
-            test_adder.add_listed_value(**bad_set_of_values)
+def test__add_to_inventory_of_listed_values_insert_after_non_final_value(test_adder):
+    new_value_id = test_adder._add_to_inventory_of_listed_values(
+        feature_id="A-3",
+        new_value_en="Central, mid-back and back",
+        new_value_ru="Средний, задне-средний и задний",
+        index_to_assign=3,
+    )
+    assert new_value_id == "A-3-3"
```

Comment:
you see that you just check the "middle" index_to_assign. But you don't try also passing the last index, e.g. `15` if the last value ID in this feature is `A-3-15`. I think that the function will fail in the current configuration. Even if I'm wrong, you should test all cases. E.g. for feature `A-3` that has values between `A-3-1` and `A-3-15`:
- `1` (edge case: first ID)
- `3`
- `10` (what if your program fails on two-digit IDs?)
- `15` (edge case: last ID)

Reply:
Done! Inserions as third and tenth are considered normal cases and insertions as first and last with explicit index are separated from them into the group of edge cases